javascripts/services/ContactService.js

```diff
@@ -0,0 +1,31 @@
+"use strict";
+
+app.service("ContactService", function ($http, $q, FIREBASE_CONFIG) {
+
+    const getUserContact = (userUid) => {
+        let contact = [];
+        return $q((resolve, reject) => {
+            $http.get(`${FIREBASE_CONFIG.databaseURL}/contacts.json?orderBy="uid"&equalTo="${userUid}"`).then((results) => {
+                let fbContacts = results.data;
+                Object.keys(fbContacts).forEach((key) => {
+                    fbContacts[key].id = key;
+                    contact.push(fbContacts[key]);
+                    console.log(contact);
+                    resolve(contact);
```

Comment:
this resolution for your contact should be outside of your looping over of your Object.keys

Reply:
@rousell I believe it's fixed now!